src/lib.rs

```diff
@@ -681,3 +691,27 @@ fn write_done(status: &OVERLAPPED_ENTRY) {
         }
     }
 }
+
+// Based on https://github.com/tokio-rs/mio/blob/13d5fc9/src/sys/windows/buffer_pool.rs
+struct BufferPool {
+    pool: Vec<Vec<u8>>,
+}
+
+impl BufferPool {
+    fn with_capacity(cap: usize) -> BufferPool {
+        BufferPool {
+            pool: Vec::with_capacity(cap),
+        }
+    }
+
+    fn get(&mut self, default_cap: usize) -> Vec<u8> {
+        self.pool.pop().unwrap_or_else(|| Vec::with_capacity(default_cap))
```

Comment:
It's intentional - I guess if the pool is empty, it makes sense to allocate a new Vec of exactly the request size (also, since the pool doesn't store the original buffer size passed in to `with_capacity`, it doesn't know any other size to use).

I agree it's a slightly odd API - it's the same as the [original version](https://github.com/tokio-rs/mio/blob/13d5fc9/src/sys/windows/buffer_pool.rs), since I tried to keep changes minimal in the hope it'd be more acceptable [upstream](https://github.com/alexcrichton/mio-named-pipes/pull/9).